lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -296,7 +297,11 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
         if (numBytes == 0)
           break;
 
-        outputData.insert(outputData.end(), &buf[0], &buf[numBytes]);
+        if (partialOutputCallback) {
+          partialOutputCallback(std::string(buf, numBytes));
+        } else {
+          outputData.insert(outputData.end(), &buf[0], &buf[numBytes]);
+        }
```

Comment:
This makes for a weird API, I wonder if we should clean up the overall API here instead of making this behavior conditional.